yt_dlp/extractor/bilibili.py

```diff
@@ -298,9 +298,13 @@ def _get_interactive_entries(self, video_id, cid, metainfo, headers=None):
 
 
 class BiliBiliIE(BilibiliBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?bilibili\.com/(?:video/|festival/\w+\?(?:[^#]*&)?bvid=)[aAbB][vV](?P<id>[^/?#&]+)'
+    _VALID_URL = r'https?://(?:www\.)?bilibili\.com/(?:video/|festival/[^/?#]+\?(?:[^#]*&)?bvid=)[aAbB][vV](?P<id>[^/?#&]+)'
 
     _TESTS = [{
+        'note': 'has `-` in its url',
+        'url': 'https://www.bilibili.com/festival/bh3-7th?bvid=BV1tr4y1f7p2&',
+        'only_matching': True,
+    }, {
```

Comment:
I prefer to keep the `only_matching` tests after the regular tests so it's easier to count through them

Reply:
Done in 86aeffae2